xComputer: Added Description

gesbeckj opened the pull request:
<!--
Thanks for submitting a Pull Request (PR) to this project. Your contribution to this project is greatly appreciated!

Please prefix the PR title with the resource name, i.e. 'xComputer: My short description'
If this is a breaking change, then also prefix the PR title with 'BREAKING CHANGE:', i.e. 'BREAKING CHANGE: xComputer: My short description'

To aid community reviewers in reviewing and merging your PR, please take the time to run through the below checklist.
Change to [x] for each task in the task list that applies to this PR.
-->
**Pull Request (PR) description**
This PR adds the ability to set the computer description through the xComputerManagement module. Along with that, the unit tests have been updated for the computer description and the examples have been updated.

**This Pull Request (PR) fixes the following issues:**
Fixes #67

**Task list:**
- [x] Change details added to Unreleased section of CHANGELOG.md?
- [X] Added/updated documentation, comment-based help and descriptions in .schema.mof files where appropriate?
- [X] Examples appropriately updated?
- [X] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [X] [Unit and (optional) Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xcomputermanagement/99)
<!-- Reviewable:end -->

msftclas commented:
@gesbeckj,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by Microsoft. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

PlagueHO commented:
Thanks @gesbeckj ! Really appreciate the contribution - always awesome to get them! I'm a bit overloaded this week, but I'll be able to review this and get it through at the end of next week (unless I can get @johlju or @bgelens to look it over :wink: ).

johlju commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 27 unresolved discussions.

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1p4XQqC0M4NjbRREI:-Kt1p4XR31efarkeDVrL:baz4fxc) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/README.md#L56)):*
> descrption

Typo 'description'

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1p9ByogXfxEaBglxX:-Kt1p9ByogXfxEaBglxY:biw6x09) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/README.md#L56)):*
> "The local computer descrption"

Could you add a more descriptive description, like "The value assigned here will set as the computer description."

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 71 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1mYuJUM10eHFnvE5P:-Kt1mYuJUM10eHFnvE5Q:bzg80ak) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L71)):*
> ```PowerShell
>         UnjoinCredential = [ciminstance]$convertToCimUnjoinCredential
>         WorkGroupName    = (Get-CimInstance -Class 'Win32_ComputerSystem').Workgroup
>         Description      = (Get-CimInstance -Class 'Win32_OperatingSystem').Description
> ```

Instead of calling `Get-CimInstance` twice for both properties `Workgroup` and `Description` could you call it once and use the object returned from `Get-CimInstance` to set both values?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 120 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1n3I4-Z-vAYm9Z_9n:-Kt1n3I4-Z-vAYm9Z_9o:b-thf3tt) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L120)):*
> Setting description

Maybe "Setting computer description..." ?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 120 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1nABIFCZ8sI7kxPj9:-Kt1nABIFCZ8sI7kxPjA:bbh4sm3) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L120)):*
> -message

Uppe 'M' in `-Message`.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 121 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1nI3N_z60C8HYXc_y:-Kt1nI3N_z60C8HYXc_z:bm1qub6) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L121)):*
> $Win32_OperatingSystem

Please change to, for example, $win32OperatingSystemCimInstance

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 303 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1nvNszth5cR_E8gaM:-Kt1nvNszth5cR_E8gaN:bctqxdz) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L303)):*
> ```PowerShell
>     if (($Name -ne 'localhost') -and ($Name -ne $env:COMPUTERNAME))
>     {
>         return $false
> ```

Non-blocking for this PR. This needs to be fixed in a separate issue : @PlagueHO it seems wrong that it returns after the first test. When running in verbose the user might want to know the status on all property tests, not just the first one (and then the Set-TargetResource will fix all). Submit an issue for that?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 307 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pkQDKPPMWuEDZPqF:-Kt1pkQDKPPMWuEDZPqG:b-mvj8ry) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L307)):*
> ```PowerShell
> 
>     Write-Verbose -Message 'Checking if description is corerect'
>     if ($Description -ne (Get-CimInstance -Class 'Win32_OperatingSystem').Description)
> ```

Please add a new line before this row.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 11 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pcJde0bLf2FaovTZ:-Kt1pcJde0bLf2FaovT_:b33m3uu) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L11)):*
> [write] string Description;

Please change to

```
[Write, Description("<same descriptive text as in README.md>")] String Description;
```

Let's make this one correct even if the other are not up to the style guideline yet.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 125 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pvyOT521wy3Js5Ib:-Kt1pvyOT521wy3Js5Ic:b-rpi6km) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L125)):*
>  Mock Get-CimInstance {[PSCustomObject]@{Description = 'This is my computer'}}

Please use named parameters, also let's try to make this a bit more readable.

```
Mock -CommandName Get-CimInstance -MockWith {
    [PSCustomObject] @{
        Description = 'This is my computer'
    }
}
```

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 126 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1sy7uY329bBJEhZkv:-Kt1sy7vYuH8l3JNlcp5:b-dtfpfi) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L126)):*
> "This is my computer"

Please use single quotes

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 127 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1t03Fdrsk2KCJ6nQj:-Kt1t03Fdrsk2KCJ6nQk:b-dtfpfi) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L127)):*
> "This is my computer"

Please use single quotes

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 130 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1rgXVODfABLrZDyhi:-Kt1rgXVODfABLrZDyhj:b-e5xsmp) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L130)):*
> ```PowerShell
>                 }
>                 It 'Should return false if description is same as specified' {
>                     Mock Get-CimInstance {[PSCustomObject]@{Description = 'This is not my computer'}}
> ```

Same as previous comment

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 131 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1t8EC7hBfnu6x8t8S:-Kt1t8EC7hBfnu6x8t8T:b-dtfpfi) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L131)):*
> "This is my computer"

Please use single quotes

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 132 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1t40DJw1cqTdu2Ru9:-Kt1t40DJw1cqTdu2RuA:b-dtfpfi) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L132)):*
> "This is my computer"

Please use single quotes

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 154 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1rkeL4UvtVZLVEXKe:-Kt1rkeL4UvtVZLVEXKf:bunbob) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L154)):*
> ```PowerShell
>                 Mock Rename-Computer {}
>                 Mock Add-Computer {}
>                 Mock Set-CimInstance {}
> ```

Please use named parameters

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 284 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1rqunfR1Baojghesb:-Kt1rquoTSHFZ1Y9yK_g:bkwa54e) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L284)):*
> illigal

Typo 'illegal'. This was a typo in code you did not write. But glad if you could fix it. :)

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 288 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1sp30wlgcfAEPbMjM:-Kt1sp30wlgcfAEPbMjN:bunbob) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L288)):*
> ```PowerShell
>                 }
>                 It 'Changes computer description in a workgroup'{
>                     Mock Get-ComputerDomain {''}
> ```

Please use named parameters

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 289 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1tC3NjAi-c7FVmhoM:-Kt1tC3NjAi-c7FVmhoN:b-xk2wfx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L289)):*
> ```PowerShell
>                 It 'Changes computer description in a workgroup'{
>                     Mock Get-ComputerDomain {''}
>                     Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso';Workgroup='Contoso';PartOfDomain=$false}}
> ```

Please use named parameters, and please make this a bit more readable (see previous comment)

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 290 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1tHiPCDrlYfyE-ez4:-Kt1tHiPCDrlYfyE-ez5:boznt2p) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L290)):*
> -Description = 'This is my computer'

Should not be an equal sign here.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 290 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1tORIQgHfVckV-mQY:-Kt1tORIQgHfVckV-mQZ:b91dcdn) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L290)):*
> -DomainName ""

Please use single quotes

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 291 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vRmm5OQpYzeiIncE:-Kt1vRmm5OQpYzeiIncF:b-ei4b9f) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L291)):*
> -Exactly 1

This should actually be `-Times 1 -Exactly`.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 294 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vpqFe2t6lWk9PuRT:-Kt1vpqFe2t6lWk9PuRU:b-xk2wfx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L294)):*
> ```PowerShell
>                 }
>                 It 'Changes computer description in a domain'{
>                     Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso.com';Workgroup='Contoso.com';PartOfDomain=$true}}
> ```

Please use named parameters, and please make this a bit more readable (see previous comment)

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 295 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vqxKDCRj8ewX65_f:-Kt1vqxKDCRj8ewX65_g:bunbob) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L295)):*
> ```PowerShell
>                 It 'Changes computer description in a domain'{
>                     Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso.com';Workgroup='Contoso.com';PartOfDomain=$true}}
>                     Mock Get-ComputerDomain {'contoso.com'}
> ```

Please use named parameters

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 296 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vtcOZbH8DmsPywFD:-Kt1vtcPNjJVFio1UQHX:b-18qfmo) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L296)):*
> $Env

$env (lower 'e')

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 297 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1w0woDVvNfsSjI2NA:-Kt1w0wpnY0YfbE9wmwS:boznt2p) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L297)):*
> -Description = 'This is my computer'

Should not be an equal sign here.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 298 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vh1s61AjuZ6C-LVq:-Kt1vh1t6Hv_tKFd6GNQ:bk86g7x) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L298)):*
> -Exactly 2

This should actually be `-Times 2 -Exactly`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: 4 of 5 files reviewed at latest revision, 26 unresolved discussions.

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1p9ByogXfxEaBglxX:-Kt2Xfi572lNeeBdfEfu:b-d8c2r8) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/README.md#L56)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "The local computer descrption"

Could you add a more descriptive description, like "The value assigned here will set as the computer description."
</blockquote></details>

Missed a 'be' in the above comment. Should have been "The value assigned here will be set as the computer description."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99)*
<!-- Sent from Reviewable.io -->

gesbeckj commented:
Review status: 1 of 5 files reviewed at latest revision, 26 unresolved discussions.

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1p4XQqC0M4NjbRREI:-Kt2UY0mjpsP5cJtteuY:b-886gsx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/README.md#L56)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> descrption

Typo 'description'
</blockquote></details>

Fixed

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1p9ByogXfxEaBglxX:-Kt2UZWFAz51OzFHPolj:b5l98v0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/README.md#L56)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Missed a 'be' in the above comment. Should have been "The value assigned here will be set as the computer description."
</blockquote></details>

Changed to be more descriptive. 

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 71 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1mYuJUM10eHFnvE5P:-Kt2Ub3DVSn5IpDamS1T:b-q5kdb4) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L71)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Instead of calling `Get-CimInstance` twice for both properties `Workgroup` and `Description` could you call it once and use the object returned from `Get-CimInstance` to set both values?
</blockquote></details>

I don't think this can be done because Workgroup and Description are in two different classes. 

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 120 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1n3I4-Z-vAYm9Z_9n:-Kt2UibIVJO99qzz_EHq:b499q7j) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L120)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> Setting description

Maybe "Setting computer description..." ?
</blockquote></details>

Changed

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 120 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1nABIFCZ8sI7kxPj9:-Kt2UmIGusQzKELSl6xU:b-886gsx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L120)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> -message

Uppe 'M' in `-Message`.
</blockquote></details>

Fixed

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 121 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1nI3N_z60C8HYXc_y:-Kt2Yeiv6_SURub8aBqt:b499q7j) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L121)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> $Win32_OperatingSystem

Please change to, for example, $win32OperatingSystemCimInstance
</blockquote></details>

Changed

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 307 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pkQDKPPMWuEDZPqF:-Kt2s_3kz1VffkjseAwN:b-896fjx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L307)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please add a new line before this row.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 11 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pcJde0bLf2FaovTZ:-Kt2satSyWnqi02vnqJX:b-z2psoy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L11)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> [write] string Description;

Please change to

```
[Write, Description("<same descriptive text as in README.md>")] String Description;
```

Let's make this one correct even if the other are not up to the style guideline yet.
</blockquote></details>

Fixed this instance and added the Description for the others to comply with style guidelines. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 125 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pvyOT521wy3Js5Ib:-Kt2rUlmU3AxpsY5LpWI:b-9keo4q) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L125)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

>  Mock Get-CimInstance {[PSCustomObject]@{Description = 'This is my computer'}}

Please use named parameters, also let's try to make this a bit more readable.

```
Mock -CommandName Get-CimInstance -MockWith {
    [PSCustomObject] @{
        Description = 'This is my computer'
    }
}
```
</blockquote></details>

I updated this and all other instances with MockWith and a PSCustom Object to that formatting for readability. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 126 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1sy7uY329bBJEhZkv:-Kt2raajNfbgHGZyv0hD:b-mkvfuo) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L126)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "This is my computer"

Please use single quotes
</blockquote></details>

I changed the double quotes to single quotes throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 127 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1t03Fdrsk2KCJ6nQj:-Kt2re8RemhrsJfPWjoS:b-mkvfuo) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L127)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "This is my computer"

Please use single quotes
</blockquote></details>

I changed the double quotes to single quotes throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 130 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1rgXVODfABLrZDyhi:-Kt2rfNLURH3v7mfIpEt:b-mkvfuo) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L130)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Same as previous comment
</blockquote></details>

I changed the double quotes to single quotes throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 131 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1t8EC7hBfnu6x8t8S:-Kt2rfwdXRyaYeQbqs52:b-mkvfuo) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L131)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "This is my computer"

Please use single quotes
</blockquote></details>

I changed the double quotes to single quotes throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 132 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1t40DJw1cqTdu2Ru9:-Kt2rgUZ_T9TI7gw72tC:b-mkvfuo) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L132)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "This is my computer"

Please use single quotes
</blockquote></details>

I changed the double quotes to single quotes throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 154 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1rkeL4UvtVZLVEXKe:-Kt2rhAbAeZnlp5cM7_i:b-wh04cs) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L154)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please use named parameters
</blockquote></details>

I updated this and all other Mock commands that I found that were not using named parameters to add the named parameter MockWith

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 284 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1rqunfR1Baojghesb:-Kt2UCuxmKw_tIHNZ-Hr:b-aqtr49) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L284)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> illigal

Typo 'illegal'. This was a typo in code you did not write. But glad if you could fix it. :)
</blockquote></details>

I fixed all "illiga" typos I could find

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 288 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1sp30wlgcfAEPbMjM:-Kt2rs-N2dYStWoezgOV:b91qqt6) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L288)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please use named parameters
</blockquote></details>

I update this and all other Mock commands to use the named parameter MockWith

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 289 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1tC3NjAi-c7FVmhoM:-Kt2rvqIFjhmWjIoh3uW:b1mhyka) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L289)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please use named parameters, and please make this a bit more readable (see previous comment)
</blockquote></details>

Named parameter and readability improved here and elsewhere in the code. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 290 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1tHiPCDrlYfyE-ez4:-Kt2ryoEvya7AxcEvfyc:by072g0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L290)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> -Description = 'This is my computer'

Should not be an equal sign here.
</blockquote></details>

Derp, I searched for that mistake throughout and fixed. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 290 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1tORIQgHfVckV-mQY:-Kt2sD6z35dv5p1gOvJ0:b-886gsx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L290)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> -DomainName ""

Please use single quotes
</blockquote></details>

Fixed

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 291 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vRmm5OQpYzeiIncE:-Kt2sFZLSDaGN8Gsaob_:b-pg7rgf) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L291)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> -Exactly 1

This should actually be `-Times 1 -Exactly`.
</blockquote></details>

I update this and the other Assert-MockCalled to use those named parameters

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 294 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vpqFe2t6lWk9PuRT:-Kt2sKbxyl8RJvhUItUr:b-b26hjb) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L294)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please use named parameters, and please make this a bit more readable (see previous comment)
</blockquote></details>

Completed throughout the code

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 295 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vqxKDCRj8ewX65_f:-Kt2sMU1OAOrFNvJQnPU:b-b26hjb) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L295)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please use named parameters
</blockquote></details>

Completed throughout the code

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 296 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vtcOZbH8DmsPywFD:-Kt2sOMYcL-thdtDZ0D6:b-yc3fen) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L296)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> $Env

$env (lower 'e')
</blockquote></details>

Fixed case

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 297 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1w0woDVvNfsSjI2NA:-Kt2sQ5I6zJ9KPFmXMJU:b-886gsx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L297)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> -Description = 'This is my computer'

Should not be an equal sign here.
</blockquote></details>

Fixed

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 298 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1vh1s61AjuZ6C-LVq:-Kt2sQrbO15vfELKd_4B:b-vlu1pz) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L298)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> -Exactly 2

This should actually be `-Times 2 -Exactly`.
</blockquote></details>

Fixed throughout to use named parameters

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99)*
<!-- Sent from Reviewable.io -->

gesbeckj commented:
I've checked through the code and updated where needed. 

---

Review status: 1 of 5 files reviewed at latest revision, 26 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-Kt2vS2n2iIIO12faG-t:b-2nntmu)*
<!-- Sent from Reviewable.io -->

johlju commented:
Thank you for fixing the style throughout, and not just your changes! Appreciate it! Less work for @PlagueHO to get this aligned to the style guideline :smile: 

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 71 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1mYuJUM10eHFnvE5P:-Kt6hXFtibPhedqqhF8u:bwvsj3g) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L71)):*
<details><summary><i>Previously, gesbeckj (James Gesbeck) wrote…</i></summary><blockquote>

I don't think this can be done because Workgroup and Description are in two different classes. 
</blockquote></details>

Yeah, sorry. I missread those class names. :) All good.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 4 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6iCAUT2Dmj4lU5cYb:-Kt6iCAUT2Dmj4lU5cYc:b-66y5no) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L4)):*
> ```
> class MSFT_xComputer : OMI_BaseResource
> {
>     [key, Description("The desired computer name")] string Name;
> ```

Since you were so kind to fix all the comments here, could you also change so that the type qualifier 'Key' is written with an upper 'K'.  

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 4 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6ivjrxoIMuaFzxp2i:-Kt6ivjrxoIMuaFzxp2j:b-93mskq) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L4)):*
> ```
> class MSFT_xComputer : OMI_BaseResource
> {
>     [key, Description("The desired computer name")] string Name;
> ```

And change the type to 'String' (upper 'S'). Throughout the schema.mof.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 5 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6iiWBIn0HNFlowGCw:-Kt6iiWBIn0HNFlowGCx:bkvy14k) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L5)):*
> write

Please change to 'Write' (upper 'W')

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 7 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6ioywpsYF1PANn0RA:-Kt6ioywpsYF1PANn0RB:b-taqdud) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L7)):*
> read

Please change to 'Read' (upper 'R')

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 125 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pvyOT521wy3Js5Ib:-Kt6j7EgzA2_kVZNXJEV:bb3sh90) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L125)):*
<details><summary><i>Previously, gesbeckj (James Gesbeck) wrote…</i></summary><blockquote>

I updated this and all other instances with MockWith and a PSCustom Object to that formatting for readability. 
</blockquote></details>

You missed `-CommandName`, like `Mock -CommandName Get-CimInstance`. Could you add that too?

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 45 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6j_d56l3pkC5Qa6UG:-Kt6j_d56l3pkC5Qa6UH:b-lcriyr) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L45)):*
> [PSCustomObject]@{

Please add a space after the type

```
[PSCustomObject] @{
...
}
```

throughout the file

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 47 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6jvinkhOkLQ2II9ac:-Kt6jvinkhOkLQ2II9ad:b-tfa2or) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L47)):*
> ```PowerShell
>                         [PSCustomObject]@{
>                             Domain = 'Contoso.com';
>                             Workgroup='Contoso.com';
> ```

Space before and after the equal sign. Throughout the file.

If you are running Visual Studio Code you could apply formatting and I think this will be aligned according to the style guideline. It's okay if it changes the style throughout.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 51 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6kRirkuf806JIjGr9:-Kt6kRirkuf806JIjGrA:baoy1oz) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L51)):*
> Mock Get-ComputerDomain -MockWith {'contoso.com'}

You could make this more readable like so.

```
Mock -CommandName Get-ComputerDomain -MockWith {
    'contoso.com'
}
```

Throughout the mocks

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 244 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6lYUE5j7EUQtobsEc:-Kt6lYUE5j7EUQtobsEd:b-1a00yp) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L244)):*
> Mock Rename-Computer -MockWith {}

For these with `-MockWith {}` you just need to use

```
Mock Rename-Computer
```

Throughout

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-Kt6kr7583wyQTIkybvw:b-b7i7k8)*
<!-- Sent from Reviewable.io -->

gesbeckj commented:
Fixes were done throughout. Any thoughts on how to best format commands with -MockWith {''} where it returns an empty string? Seems unnecessary to put these on multiple lines. 

---

Review status: 3 of 5 files reviewed at latest revision, 11 unresolved discussions.

---

*[README.md, line 56 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1p4XQqC0M4NjbRREI:-Kt2vZ7D9uwXs0gO0hMY:b-896fix) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/README.md#L56)):*
<details><summary><i>Previously, gesbeckj (James Gesbeck) wrote…</i></summary><blockquote>

Fixed
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 71 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1mYuJUM10eHFnvE5P:-Kt2vavXjA3CK28aPfIw:bmfxq1d) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L71)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Yeah, sorry. I missread those class names. :) All good.
</blockquote></details>

Done. 

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 4 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6iCAUT2Dmj4lU5cYb:-Kt7M-XDOZwNvK-ZiXgn:b-lx4fj3) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L4)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Since you were so kind to fix all the comments here, could you also change so that the type qualifier 'Key' is written with an upper 'K'.  
</blockquote></details>

Fixed. 

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 4 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6ivjrxoIMuaFzxp2i:-Kt7M8wzgnehPOfYLThA:b-6v92t4) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L4)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

And change the type to 'String' (upper 'S'). Throughout the schema.mof.
</blockquote></details>

Fixed throughout the schema. 

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 5 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6iiWBIn0HNFlowGCw:-Kt7MAfadij4XOpkQvBK:b-u0v79u) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L5)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> write

Please change to 'Write' (upper 'W')
</blockquote></details>

Fixed through the schema. 

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof, line 7 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6ioywpsYF1PANn0RA:-Kt7MBykTHZ7nJBgkFnE:b-6v92t4) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/DSCResources/MSFT_xComputer/MSFT_xComputer.schema.mof#L7)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> read

Please change to 'Read' (upper 'R')
</blockquote></details>

Fixed throughout the schema. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 125 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1pvyOT521wy3Js5Ib:-Kt7MrkMr8YR99Ty5YW8:b-lw033f) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/Tests/Unit/MSFT_xComputer.Tests.ps1#L125)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

You missed `-CommandName`, like `Mock -CommandName Get-CimInstance`. Could you add that too?
</blockquote></details>

Good catch, fixed. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 45 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6j_d56l3pkC5Qa6UG:-Kt7MKdzBd0mv2r1olYi:b-toyp04) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L45)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> [PSCustomObject]@{

Please add a space after the type

```
[PSCustomObject] @{
...
}
```

throughout the file
</blockquote></details>

Fixed throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 47 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6jvinkhOkLQ2II9ac:-Kt7MLnxhLhcpJ-cFt5z:b-rfleds) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L47)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Space before and after the equal sign. Throughout the file.

If you are running Visual Studio Code you could apply formatting and I think this will be aligned according to the style guideline. It's okay if it changes the style throughout.
</blockquote></details>

I didn't realize Visual Studio Code could do that. That worked amazing!! Formatting fix has been done throughout the file. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 51 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6kRirkuf806JIjGr9:-Kt7McLoE_qaO-hYZKEL:bzav25q) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L51)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> Mock Get-ComputerDomain -MockWith {'contoso.com'}

You could make this more readable like so.

```
Mock -CommandName Get-ComputerDomain -MockWith {
    'contoso.com'
}
```

Throughout the mocks
</blockquote></details>

Done throughout the mocks. 

Any suggestions for how the formatting should be done for the -MockWith {''}? Seems silly to have those extend to multiple lines as well. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 244 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6lYUE5j7EUQtobsEc:-Kt7MteCjP743zoKtLCi:b7mse29) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L244)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> Mock Rename-Computer -MockWith {}

For these with `-MockWith {}` you just need to use

```
Mock Rename-Computer
```

Throughout
</blockquote></details>

Removed -MockWith {} throughout. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-Kt7Mzvuk5SPiUIRe-cC:b-kpanun)*
<!-- Sent from Reviewable.io -->

johlju commented:
See review comment below. Just a few more comments for you to fix then it looks good to me. 

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 47 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6jvinkhOkLQ2II9ac:-Kt7WVj-jZ5KEnIaHvPR:b-6aix48) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L47)):*
<details><summary><i>Previously, gesbeckj (James Gesbeck) wrote…</i></summary><blockquote>

I didn't realize Visual Studio Code could do that. That worked amazing!! Formatting fix has been done throughout the file. 
</blockquote></details>

Yeah, it's pretty neat. :) There is a vscode project settings file in this repo making sure the formatting follow the guideline as close as it can. Thanks for formatting the file!

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 51 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6kRirkuf806JIjGr9:-Kt7Wp1qk-wHQWDUifHM:bhhnzwx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L51)):*
<details><summary><i>Previously, gesbeckj (James Gesbeck) wrote…</i></summary><blockquote>

Done throughout the mocks. 

Any suggestions for how the formatting should be done for the -MockWith {''}? Seems silly to have those extend to multiple lines as well. 
</blockquote></details>

Actually for consistency I would have done the same for them, even if it seems silly with just `''` on a single row . But I leave this on up to @PlagueHO. Non-blocking.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 34 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt7W2OZnk5Km2DLyDyI:-Kt7W2O_dyO-WhwJ-PU7:b-8wghj2) ([raw file](https://github.com/powershell/xcomputermanagement/blob/465556c2fe290b35bc51808e734e279184af206d/Tests/Unit/MSFT_xComputer.Tests.ps1#L34)):*
> ```PowerShell
>             $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList 'USER', $SecPassword
>             $NotComputerName = if ($env:COMPUTERNAME -ne 'othername')
>             {'othername'
> ```

Please make sure the opening brace is alone on the row. Here an on the else below (only ones I saw).

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 143 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt7XPWhH_o2JuKF6-hX:-Kt7XPWhH_o2JuKF6-hY:b-onkklz) ([raw file](https://github.com/powershell/xcomputermanagement/blob/465556c2fe290b35bc51808e734e279184af206d/Tests/Unit/MSFT_xComputer.Tests.ps1#L143)):*
> ```PowerShell
>                         }
>                     }
>                     Mock -CommandName Get-ComputerDomain -MockWith {'contoso.com'}
> ```

Missed splitting this one into more readable form.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-Kt7YVqVbgk_UCRk8Nyz:b-36bt9q)*
<!-- Sent from Reviewable.io -->

gesbeckj commented:
Fixes are made. 

---

Review status: 4 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 47 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6jvinkhOkLQ2II9ac:-Kt7hFuL4QhvmGT-Zu9r:bb6un1j) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L47)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Yeah, it's pretty neat. :) There is a vscode project settings file in this repo making sure the formatting follow the guideline as close as it can. Thanks for formatting the file!
</blockquote></details>

Done

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 51 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6kRirkuf806JIjGr9:-Kt7h79WZfxY2_mIHHFc:bjs3y1v) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L51)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Actually for consistency I would have done the same for them, even if it seems silly with just `''` on a single row . But I leave this on up to @PlagueHO. Non-blocking.
</blockquote></details>

Consistency is a good point. I'll update them throughout. 

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 34 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt7W2OZnk5Km2DLyDyI:-Kt7hDWlhlEwj93Ln5AT:b-886gsx) ([raw file](https://github.com/powershell/xcomputermanagement/blob/465556c2fe290b35bc51808e734e279184af206d/Tests/Unit/MSFT_xComputer.Tests.ps1#L34)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please make sure the opening brace is alone on the row. Here an on the else below (only ones I saw).
</blockquote></details>

Fixed

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 143 at r4](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt7XPWhH_o2JuKF6-hX:-Kt7hAZXeO2cGKxD7_gE:b-lx4fj3) ([raw file](https://github.com/powershell/xcomputermanagement/blob/465556c2fe290b35bc51808e734e279184af206d/Tests/Unit/MSFT_xComputer.Tests.ps1#L143)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Missed splitting this one into more readable form.
</blockquote></details>

Fixed. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-Kt7m-SqAGygZXydCl9X:b1cx9yt)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

@PlagueHO can you look this over and merge?

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-Kt7rKfvq7y244ZKWrpq:bw2mtaq)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Great job @gesbeckj and thanks for reviewing @johlju. Just one minor issue there to fix I think. Then I'll merge.

---

Reviewed 2 of 4 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 303 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt1nvNszth5cR_E8gaM:-KtAkfNLyS3zANPBvsfW:bey5m7z) ([raw file](https://github.com/powershell/xcomputermanagement/blob/7385faa3b60ee1681d25b65fb71baf06f4b2f2ac/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L303)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Non-blocking for this PR. This needs to be fixed in a separate issue : @PlagueHO it seems wrong that it returns after the first test. When running in verbose the user might want to know the status on all property tests, not just the first one (and then the Set-TargetResource will fix all). Submit an issue for that?
</blockquote></details>

@Johlju - yep - you're quite right. I'll submit an issue. We can leave it as is in this PR but should fix it.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 120 at r5](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-KtAjvzw8-GiiWz_64qb:-KtAjvzw8-GiiWz_64qc:b-8b9cbg) ([raw file](https://github.com/powershell/xcomputermanagement/blob/71cc59ea0670a877fc27e32d3fb6c19e37d26473/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L120)):*
> ```PowerShell
>     }
> 
>     Write-Verbose -Message "Setting computer description to '$($Description)'."
> ```

I'm just wondering: What will happen here if the $Description is not set - e.g. if you're just setting the computer name, domain or workgroup? Won't this cause the Description to be blanked?

It might be better to wrap this in:
```powershell
if ($PSBoundParameters.ContainsKey('Description') {
   Write-Verbose -Message "Setting computer description to '$($Description)'."
   $win32OperatingSystemCimInstance = Get-CimInstance -ClassName Win32_OperatingSystem
   $win32OperatingSystemCimInstance.Description = $Description
   Set-CimInstance -InputObject $win32OperatingSystemCimInstance
}
```

---

*[Tests/Unit/MSFT_xComputer.Tests.ps1, line 51 at r3](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-Kt6kRirkuf806JIjGr9:-KtAkxOHGvQRKIfgnh6J:b-r6kov5) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4b38c3f4201bc8e7525fb1f2b12cd575a6783b28/Tests/Unit/MSFT_xComputer.Tests.ps1#L51)):*
<details><summary><i>Previously, gesbeckj (James Gesbeck) wrote…</i></summary><blockquote>

Consistency is a good point. I'll update them throughout. 
</blockquote></details>

I'm going to go through all these files anyway and make them HQRM, and at the same time I'll correct all these Mock styles as well. Agree that it should be done but not-blocking.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-KtAl9tHWjOagOvqFOEZ:bxf52oi)*
<!-- Sent from Reviewable.io -->

gesbeckj commented:
I've added the if statement around both the Set-TargetResource and Test-TargetResource to avoid blanking out the description in the case where the description is not specified in the DSC configuration but is configured on the computer already. 

---

Review status: 3 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 120 at r5](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-KtAjvzw8-GiiWz_64qb:-KtEH6s2fMxewnGXdoOk:bkc9eef) ([raw file](https://github.com/powershell/xcomputermanagement/blob/71cc59ea0670a877fc27e32d3fb6c19e37d26473/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L120)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I'm just wondering: What will happen here if the $Description is not set - e.g. if you're just setting the computer name, domain or workgroup? Won't this cause the Description to be blanked?

It might be better to wrap this in:
```powershell
if ($PSBoundParameters.ContainsKey('Description') {
   Write-Verbose -Message "Setting computer description to '$($Description)'."
   $win32OperatingSystemCimInstance = Get-CimInstance -ClassName Win32_OperatingSystem
   $win32OperatingSystemCimInstance.Description = $Description
   Set-CimInstance -InputObject $win32OperatingSystemCimInstance
}
```
</blockquote></details>

That's a good point for that behavior when it isn't specified. I wrapped both this with the if statement as well as the Test function so DSC will not report it is not in compliance because the description was not specified and a description currently exists. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-KtEHSQYR96I1Fgl4Jdj:b492cx)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Thanks @gesbeckj - fantastic job sir! Just one tiny style nitpick (just need one linefeed) :grin:

---

Reviewed 1 of 5 files at r1, 1 of 2 files at r4, 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 126 at r6](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-KtLZGMu17Sv9IzwMsD7:-KtLZGMv-25edsTU_pUm:b3zuwz0) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4590405b0d46875e614fc240f056c325fee33be0/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L126)):*
> ```PowerShell
>         $win32OperatingSystemCimInstance.Description = $Description
>         Set-CimInstance -InputObject $win32OperatingSystemCimInstance
>     }
> ```

Tiny style nitpic - can you add a line feed after the } ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-KtLZjdA7eViFF6BMU_n:b45cjbm)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 315 at r6](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-KtLZxMpNvDFzqfCrxHK:-KtLZxMpNvDFzqfCrxHL:b-i6lpsy) ([raw file](https://github.com/powershell/xcomputermanagement/blob/4590405b0d46875e614fc240f056c325fee33be0/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L315)):*
> ```PowerShell
>             return $false
>         }
>     }
> ```

Tiny style nitpic - can you add a line feed after the } ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Awesome! Thank you @gesbeckj

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/99#-:-KtQYunA_7FAAVfkiQzI:b-uw3w22)*
<!-- Sent from Reviewable.io -->
